CommcareTranslationChecker/CommcareTranslationChecker.py

```diff
@@ -207,24 +232,133 @@ def get_invalid_format_tags(base_column_value, output_column_value):
+        if base_column_matches_count != output_column_matches_count:
+            invalid_format_tags.append(tag)
+
     return invalid_format_tags
 
 
-def checkRowForMismatch(row, columnDict, baseColumnIdx=None, ignoreOrder=False, wsOut=None, mismatchFlagIdx=None,
-                        outputMismatchTypesFlag=False, skipFormatCheckFlag=False, formatCheckCharacters=None,
-                        formatCheckCharactersAdd=None, verbose=False):
+def fix_block_tags_mismatch(baseText, outputText):
+    """
+        Fixes mismatches for headings, ordered list and unordered list
+
+        Input:
+        baseText(str): base column text
+        outputText(str): Comparing column text
+
+        :return: Fixed output text
+        """
+    baseTextLines = baseText.splitlines()
+    outputTextLines = outputText.splitlines()
+    fixed_output_text = []
+
+    # If line count in base text and output text doesn't match,
+    # return outputText as is
+    if len(baseTextLines) != len(outputTextLines):
+        return outputText
+
+    try:
+        # Compare each line of baseText with outputText,
+        # if mismatch occurs, fix it
+        for index in range(len(baseTextLines)):
+            baseTextLine = baseTextLines[index]
+            outputTextLine = outputTextLines[index]
+            regex_match_found = False
+
+            for tag in BLOCK_FORMATTING_TAGS:
+                base_tag_char_list = []
+                if len(re.findall(tag, baseTextLine)) > 0:
+                    regex_match_found = True
+
+                    for char in baseTextLine:
+                        base_tag_char_list.append(char)
+                        if char == ' ':
+                            break
+
+                    for position in range(len(base_tag_char_list)):
+                        if outputTextLine[position] != base_tag_char_list[position]:
+                            break
+
+                    if position == len(base_tag_char_list):
+                        fixed_output_text.append(outputTextLine)
+                        break
+                    else:
+                        fixed_output_text.append(''.join(base_tag_char_list) + outputTextLine[position:])
+                        break
+
+            if regex_match_found is False:
+                fixed_output_text.append(outputTextLine)
+        return '\n'.join(fixed_output_text)
+    except:
+        # If exception occurs while comparison, return outputText value received
+        return outputText
+
+
+def fix_output_tags_mismatch(baseText, mismatchTypes, curOutputValueList, missingValueList, extraValueList):
+    currentValue = baseText
+
+    # Fix the case where there is 1 missing output tag value and 1 new output tag value
+    # Assumption: 1 existing output tag value has been changed/translated mistakenly
+    if len(missingValueList) == 1 and len(extraValueList) == 1:
+        currOutputTag = "<output value=\"" + extraValueList[0] + "\"/>"
```

Comment:
if I remember I spent quite some time to make the camel case underscore case in this library but I might have missed places, don't let that get you here.